plugins/adminchat.sma

```diff
@@ -216,23 +216,18 @@ public cmdSayAdmin(id)
 	log_amx("Chat: ^"%s<%d><%s><>^" chat ^"%s^"", name, userid, authid, message[1])
 	log_message("^"%s<%d><%s><>^" triggered ^"amx_chat^" (text ^"%s^")", name, userid, authid, message[1])
 	
-	if (is_user_admin(id)) // no diff here if admins have g_AdminChatFlag access or not, but we don't want to print "PLAYER"
-		format(message, charsmax(message), "(%l) %s :  %s", "ADMIN", name, message[1])
-	else
-		format(message, charsmax(message), "(%l) %s :  %s", "PLAYER", name, message[1])
-
 	get_players(players, inum, "ch")
 	
-	for (new i = 0; i < inum; ++i)
+	for (new bool:is_sender_admin = !!is_user_admin(id), i = 0; i < inum; ++i)
 	{
 		pl = players[i]
+
 		if (pl == id || get_user_flags(pl) & g_AdminChatFlag)
 		{
-			SetGlobalTransTarget(pl)
-			client_print(pl, print_chat, "%s", message)
+			client_print(pl, print_chat, "(%L) %s :  %s", pl, is_sender_admin ? "ADMIN" : "PLAYER", name, message[1])
```

Comment:
Should be `%l`.

You use `%L` when you explicit the player's id.

You use `%l` when the native already passes the player's id such as `client_print(id, ...`.  You can do that because internally the player's id is set globally before the string is actually formatted. When the parser will check `%l`, it will use the saved value. Which means for native like formatting native `format`, you can use before `SetGlobalTransTarget` to set globally the player's id and then you can use `%l` without issue.

Reply:
I didn't know that. Thanks, I changed it.